doc/src/sgml/xlstproc_depfile_wrapper.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+import os
+import shutil
+import subprocess
+import sys
+import argparse
+from unittest import result
+
+parser = argparse.ArgumentParser(
+    description='generate dependency file for docs')
+
+parser.add_argument('--basename', type=str, required=False, nargs='+')
+parser.add_argument('--depfile', type=str, required=False)
+parser.add_argument('-i', '--input', type=str, nargs='+')
+parser.add_argument('-o', '--output', type=str, required=True)
+parser.add_argument('--path', type=str, required=True)
+parser.add_argument('--xsltproc', type=str, required=True)
+
+# since both flags and inputs can have multiple arguments
+# and also flags can have arguments starts with `--`,
+# flags parsed as a remaining args
+args, flags = parser.parse_known_args()
+# first element is `--flags`, discard it
+flags = flags[1:]
+
+path = os.path.realpath(args.path)
+
+if args.depfile:
+    depfile = os.path.realpath(args.depfile)
+    depfile_result_list = []
+
+    command = [args.xsltproc, '--path', path, '--load-trace', '-o', \
+                args.output] + flags + args.input
+    # --load-trace flag displays all the documents
+    # loaded during the processing to stderr
+    res = subprocess.run(command, stderr=subprocess.PIPE, check=True)
+    # make dependencies list
+    dependencies = res.stderr.decode("utf-8").split('\n')
+
+    # get line start from basenames
+    line_start = ''
+    for name in args.basename:
+        line_start = line_start + \
+            os.path.join(os.path.dirname(args.output) + os.sep + os.path.basename(name)) + ' '
```

Comment:
I suspect pycodestyle will complain about an overly-long line here (and perhaps a few other places).